### DISCUSSION TITLE
Clarify global components

### DISCUSSION THREAD

**Discussion by leoj3n** [vuejs/vitepress#561] [Q&A] [answered]
As someone unfamiliar with `vue`, on that page:

https://vitepress.vuejs.org/guide/global-component.html

For `ClientOnly`, it is not clear to me what is meant by:

> (for example, contain custom directives)

What is meant by custom directive?

As well as for `OutboundLink`:

> In VitePress, this component has been followed by every external link.

What is meant by "has been followed by every external link"?

If someone would help explain these two sentences I would submit a PR with improved descriptions.

**Comment by WebMechanic** [comment, marked as answer]
from https://vuejs.org/guide/reusability/custom-directives.html
> Vue also allows you to register your own custom directives. ... 
> Custom directives ... are mainly intended for reusing logic that involves low-level DOM access on plain elements.

_Built in directives_ include things such as `v-if` `v-else` that conditionally render DOM elements or others to "inject" data into their attributes and of course much more. If you create your own *custom directives* they should follow certain rules to work with SSR.

`OutboundLink` is a small component that simply renders an SVG icon **after** an external link, that is it "follows" the generated `<a>` link element.
That's indeed a weired description that tells nothing about the icon, plus there are other icons, too.

**Reply by leoj3n** [reply]
Thanks! I've submitted a PR: https://github.com/vuejs/vitepress/pull/568

**Comment by WebMechanic** [comment]
for the curious https://github.com/vuejs/vitepress/tree/main/src/client/theme-default/components/icons
